### PR TITLE
DaemonSet e2e: Update image and rolling upgrade test timeout

### LINKED ISSUE #71666: Conformance test for DaemonSet RollingUpdate to rigid on timeout
When running conformance testing on a cluster with 8 nodes schedulable for jobs, the conformance test for rolling updates on daemon sets fails due to timeout.

Test name: `Daemon set [Serial] should update pod when spec was updated and update strategy is RollingUpdate [Conformance]`

Test code is here:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/daemon_set.go#L326-L369
It utilizes a hardcoded timeout limit of 5m.

**What happened**:
The logs seem to indicate nothing wrong other than the time it takes to complete the rolling restart. The rolling restart timeout is hardcoded at 5m which doesn't seem to be enough time for large enough clusters.

**What you expected to happen**:
I would expect the timeout to either be sliding or for the check to be kicking the can; ensuring that the update is proceeding by at least one node every minute or two.

**How to reproduce it (as minimally and precisely as possible)**:
Requires a cluster with at least 8 schedulable nodes; more would make the timeout more certain.

**Anything else we need to know?**:

I was trying to see what k8s was testing itself with and it seems (according to this log) that the test is run against a cluster with 4 schedulable nodes and takes ~3m which would line up with what I'm seeing (8 nodes for ~6m, 1 more than given by the hardcoded deadline)

https://storage.googleapis.com/kubernetes-jenkins/logs/ci-cri-containerd-e2e-gci-gce-serial/1555/build-log.txt

Seems like there are a few options:
 - use a "kick the can" strategy to ensure the timeout scales
 - use a timeout calculated from the number of schedulable nodes
 - (mitigation) increase the max number of down nodes so the update can occur faster

**Environment**:
- Kubernetes version (use `kubectl version`):
From test logs (cluster no longer available):
```
INFO: e2e test version: v1.12.1
INFO: kube-apiserver version: v1.12.2-1+619f4e6f7f010f
```

/sig testing
/kind bug

### PR DESCRIPTION
**Pull request by alexbrand**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**: 
This PR removes an unnecessary sleep from the serve-hostname binary that is used in conformance tests.

The reason for this change is that the introduction of this 60 second sleep has negatively impacted the DaemonSet rolling upgrade conformance test. 

Specifically, it prevents the `serve-hostname` pods from terminating gracefully, and thus each pod takes at least 30 seconds to terminate (due to the default 30 second termination grace period). 

In large clusters, the conformance test fails because rolling all pods of the DaemonSet takes longer than timeout configured in the conformance test (5 minutes).

**Which issue(s) this PR fixes**:

Fixes #71666 

**Special notes for your reviewer**:
I have updated all the references to this image that I could find using code search. Not sure if I missed any.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/testdata/pod-with-large-name.yaml [hunk 1]
```diff
@@ -8,4 +8,4 @@ metadata:
 spec:
   containers:
   - name: kubernetes-serve-hostname
-    image: gcr.io/kubernetes-e2e-test-images/serve-hostname-amd64:1.1
+    image: gcr.io/kubernetes-e2e-test-images/serve-hostname-amd64:1.3
```
Comment (ixdy) [inline hack/testdata/pod-with-large-name.yaml]: putting the reference updates into a separate commit might make this easier to cherrypick to release branches.

Comment (alexbrand) [reply hack/testdata/pod-with-large-name.yaml]: Thanks @ixdy. Will move the updates to a separate commit.

**Comment by alexbrand** [comment]
@freehan I think adding a flag is a possibility, but another potential route is just updating the DaemonSet test instead. I tagged you in the original issue as there is some more discussion happening there. Would love your insight and thoughts.

**Comment by alexbrand** [comment]
Updated the title to reflect the new approach for this fix. This is a more direct fix to the problem being observed in #71666. 

Instead of changing the `serve-hostname` image which is depended on by other tests, we fix the DaemonSet test directly. 

@BenTheElder, @ixdy, @freehan PTAL

**Comment by ixdy** [comment]
/approve

I'll let @freehan apply lgtm.

File: test/e2e/apps/daemon_set.go [hunk 2]
```diff
@@ -329,6 +329,10 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 		framework.Logf("Creating simple daemon set %s", dsName)
 		ds := newDaemonSet(dsName, image, label)
 		ds.Spec.UpdateStrategy = apps.DaemonSetUpdateStrategy{Type: apps.RollingUpdateDaemonSetStrategyType}
+		// Set the termination grace period to 1 second. This is necessary because the
+		// serve-hostname binary sleeps for 1 minute after receiving SIGTERM.
+		terminationPeriod := int64(1)
+		ds.Spec.Template.Spec.TerminationGracePeriodSeconds = &terminationPeriod
```
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: This fix should work, but DaemonSet test shouldn't be tied to the serve-hostname image. We can simply change the image we use in this test:
```
image := framework.ServeHostnameImage
```
to something else (can pick from [this list](https://github.com/kubernetes/kubernetes/blob/a2a5bd03fdae6ac45509560157f4c24d1c001e86/test/utils/image/manifest.go#L93)), such as nginx.

We can also increase the timeout based on number of nodes, as this affects the number of replicas a DaemonSet creates.

Comment (alexbrand) [reply test/e2e/apps/daemon_set.go]: That is a good point. Should we take that route instead? Will take a look at the list of images.

Comment (janetkuo) [reply test/e2e/apps/daemon_set.go]: Yes, thanks for looking into this!

Comment (alexbrand) [reply test/e2e/apps/daemon_set.go]: Should we update the image used throughout all DaemonSet tests, or just the one used during the DaemonSet update test?

Comment (janetkuo) [reply test/e2e/apps/daemon_set.go]: Updating the image used throughout all DaemonSet tests (i.e. change `image := framework.ServeHostnameImage`) is fine. Which image we pick doesn't matter. 

We can scale update timeout based on # of nodes too, maybe something like `5 mins + 1 min * # of nodes`.

**Comment by alexbrand** [comment]
Updated the image, and made the timeout a function of the number of nodes in the cluster. PTAL.

**Review by janetkuo — APPROVED** [review]

**Comment by BenTheElder** [comment]
thank you!

**Comment by janetkuo** [comment]
We need to cherrypick this to older releases as well.

**Comment by janetkuo** [comment]
Sent cherrypick PRs: 
https://github.com/kubernetes/kubernetes/pull/74820
https://github.com/kubernetes/kubernetes/pull/74822

**Pull request merged**